database/mongodb/README.md

```diff
@@ -0,0 +1,135 @@
+
+At the end of the `%post` section, access is also given to the specific user in the specific group to the database folder in which data is stored.
+
+Then we define a section called `%startscript` which is the section that is executed everytime a container instance is started. In here we find the commands that define a restart of the MongoDB daemon: first a stop of the daemon is executed using `mongod --repair` and then the daemon is started using the `mongod` command. After this, the service can be started with `mongo --host $HOSTNAME:$PORT` in the next line. See that `$HOSTNAME` and `$PORT` correspond to the environment variables we defined and exported at the beginning in the `%environment` section.
+
+Another important feature to notice is that you can also refer to these environment variables on runtime with the prefix `$SINGULARITYENV_` , following this example `$SINGULARITYENV_HOSTNAME` and `$SINGULARITYENV_PORT` can be referenced from runtime.
+
+To run the definition file and build our container, we will need to:
+
+```
+$ sudo singularity build mongodb.sif mongodb.def
+```
+
+This will generate a `mongodb.sif` Singularity image. From there, you will need to bind the `/data/db` folder from the host before running the container. This folder is where the data from the database such as logs and/or journals are stored. Open a command line and then just run:
+
+```
+$ mkdir -p mongodb/db
+$ singularity shell -B mongodb/db:/data/db mongodb.sif
+```
+
+With `-B` we bind the `mongodb/db` folder on the host to the `data/db` folder that is referenced from the container.
+
+After this we are ready to run an instance, for this we run the following command:
+
+```
+$ sudo singularity instance start mongodb.sif mongo
+```
+
+We can see there the `mongo` server running and the shell of the `mongo` server waiting for input. From here you can start setting up your database by, for example, adding a new user.
+
+To add a new user in the database and give it a specific role, you can do so by:
+
+First, shell into the container and run:
+
+```
+$ sudo singularity shell mongodb.sif
+
+mongodb> mongod --repair
+mongodb> mongod
+mongodb> mongo
+```
+
+That will start the MongoDB shell and from there you can create a user for example:
+
+```
+> use admin
+
+> db.createUser(
+  {
+    user: "myUserAdmin",
+    pwd: "test",
+    roles: [ { role: "userAdminAnyDatabase", db: "admin" } ]
+  }
+)
+
+```
+
+In this example, we have created a user called `myUserAdmin` with a `userAdminAnyDatabase` role and with access to the `admin` database, this user has also as a password (`pwd`) the value `test`.
+
+After this you can exit the mongo shell and then `shell` into the container to access the database with the new created user.
+
+You can do so by:
+
+```
+$ sudo singularity shell mongodb.sif
```

Comment:
This isn't bind mounting in a directory that can be written to within the container from this command.